rpc/types/src/macros.rs

```diff
@@ -152,22 +222,7 @@ macro_rules! define_request {
         // Any doc comments, derives, etc.
         $( #[$attr:meta] )*
         // The response type.
-        $t:ident {}
-    ) => {
-        $( #[$attr] )*
-        ///
-        /// This request has no inputs.
-        pub type $t = ();
-    };
-
-    //------------------------------------------------------------------------------
-    // This branch of the macro expects fields within the `{}`,
-    // and will generate a `struct`
-    (
-        // Any doc comments, derives, etc.
-        $( #[$attr:meta] )*
-        // The response type.
-        $t:ident {
+        $t:ident $(($restricted:ident $(, $empty:ident)?))? {
```

Comment:
Empty requests being aliased to `()` unfortunately has to go since each request must have a concrete type so we can implement `RpcCall` on it.

Reply:
Although it's worth noting `()` aliases didn't matter for `cuprate_rpc_interface` usage.